### ISSUE TITLE
Android TV remote control does not work

### ISSUE THREAD

**Issue by SuperUserNameMan** [godotengine/godot#4422]
**Operating system or device:**

Godot 2.0.2_stable official, Win64
Android-TV 5.1 lollipop

**Issue description** 

Pressing the AndroidTV remote control buttons does not generate any input events.

The tv-remote "back" button exits the godot game, but the tv-remote DPAD and OK (DPAD_CENTER) don't work.
Pressing tv-remote numpad does not work neither.

According to the AndroidTV doc, the tv-remote is supposed to work as a DPAD, and the "OK" button (at the center of the tv-remote DPAD) should return a KEYCODE_DPAD_CENTER that is generally treated as a BUTTON_A. 

On the other hand, when I plug an USB XBOX 360 controller into the TV, the game controller works pretty well.

Just in case, here you have some resources about android game controllers and tv-remote :
https://github.com/chuckjr/AndroidTV_TVInputDeviceTest/blob/74cf52e7f02d24efba770566c8aa0ecb352164a1/app/src/main/java/com/stinkless/tvinputdevicetest/activity/MainActivity.java
http://developer.android.com/training/game-controllers/controller-input.html
https://developer.amazon.com/appsandservices/community/post/TxDMC4ZG5XCTCN/10-Tips-for-Remote-and-Controller.html

**Steps to reproduce:**

Run the 

**Link to minimal example project** (optional but very welcome):

```
test.scn
    Control (+script)
        Label
```

```
extends Control

func _ready():
    set_process_input(true)

func console(string):
    var label = get_node("Label")
    label.set_text( label.get_text() + " ; " + string )

func _input(event):
    console(""+str(event.type)+", "+str(event.device)+", "+str(event.ID) )

```

**Comment by SuperUserNameMan** [comment]
Duno if it would be useful information, but according to the official "joystick" demo, the name of the TV remote is "TPV-RC6".

(Pressing the remote buttons does nothing into the joystick demo.)

**Comment by SuperUserNameMan** [comment]
Also, since my TV is Android 5.1 (loolipoop), I managed to recompile android export templates using Android SDK and Android build tools version 21, and java 8 instead of version 19 and 6 like stated into the documentation.

This made no difference.

Also, I tried to modify a little de source code to try to see if something magic would happen, and, I DONT REMEMBER WITH WHICH VERSION, the DPAD_LEFT of the remote worked jsut once, but was interpreted as a DPAD_LOW. That was the only button that worked, and I could not reproduce.

**Comment by Calinou** [comment]
First of all thank you for your report and sorry for the delay.

We released Godot 3.0 in January 2018 after 18 months of work, fixing many old issues either directly, or by obsoleting/replacing the features they were referring to.

We still have hundreds of issues whose relevance/reproducibility needs to be checked against the [current stable version](https://godotengine.org/download), and that's where you can help us.
**Could you check if the issue that you described initially is still relevant/reproducible in Godot 3.0 or any newer version, and comment about it here?**

For bug reports, please also make sure that the issue contains detailed steps to reproduce the bug and, if possible, a zipped project that can be used to reproduce it right away. This greatly speeds up debugging and bugfixing tasks for our contributors.

Our Bugsquad will review this issue more in-depth in 15 days, and potentially close it if its relevance could not be confirmed.

Thanks in advance.

*Note: This message is being copy-pasted to many "stale" issues (90+ days without activity). It might happen that it is not meaningful for this specific issue or appears oblivious of the issue's context, if so please comment to notify the Bugsquad about it.*

**Comment by akien-mga** [comment]
> Our Bugsquad will review this issue more in-depth in 15 days, and potentially close it if its relevance could not be confirmed.

As there was no update since the previous post, we close this issue.

If it is still relevant in the way it was described and discussed above, please comment to ask for it to be reevaluated. If it is only partly relevant, or if the original issue has changed, it would be better to open a new issue (potentially referring to this one) focused on the current state.

**Comment by kmanpro** [comment]
I am experiencing the same problem with the Amazon Fire TV. No events are registered for the buttons pressed.

I am using the latest Godot 3.05

**Comment by mustii82** [comment]
any news ?

**Comment by fillibuster77** [comment]
Hi,

I have the same issues here. I tried with a Fire TV Stick and an Android TV device. Any suggestions?

**Comment by lawnjelly** [comment]
This may well be fixed by #29948. I think my tv box remote control was responding to input after the fix. I.e. grab a recent nightly build and try.

However note that there is a separate but related issue for cursor keys (and perhaps some other keys). These are all being returned as unhandled to the OS, so the OS may decide to do other actions, which caused crash for cursor keys on my devices (it appeared to be doing something with audio, perhaps a default android mapping). I have been intending to create an issue about this.

Keys that are being used by Godot should probably be returned as handled to prevent this. However, due to Godot design it isn't possible (afaik) to know ahead of time which keys we will be using, so there may need to be some compromise, like we report all keys as handled except for e.g. back button, volume buttons.

**Comment by akien-mga** [comment]
Assuming fixed by #29948.

Please comment if you can still reproduce this issue.
